source/fab/tasks/fortran.py

```diff
@@ -359,13 +366,26 @@ def __init__(self, reader: TextReader, database: SqliteStateDatabase):
     _end_block_pattern: Pattern = re.compile(_end_block_re, re.IGNORECASE)
     _use_pattern: Pattern = re.compile(_use_statement_re, re.IGNORECASE)
 
-    def run(self):
+    def run(self, artifacts: List[Artifact]) -> List[Artifact]:
         logger = logging.getLogger(__name__)
 
+        if len(artifacts) == 1:
+            artifact = artifacts[0]
+        else:
+            msg = ('Fortran Analyser expects only one Artifact, '
+                   f'but was given {len(artifacts)}')
+            raise TaskException(msg)
+
+        reader = FileTextReader(artifact.location)
```

Comment:
Artefacts may gain the ability to return a reader object directly, in the future.

Reply:
That's a nice idea, similar to pushing the hashing inside the object